cmd/zar/ls/command.go

```diff
@@ -29,25 +32,74 @@ func init() {
 
 type Command struct {
 	*root.Command
+	root  string
+	lflag bool
 }
 
 func New(parent charm.Command, f *flag.FlagSet) (charm.Command, error) {
 	c := &Command{Command: parent.(*root.Command)}
+	f.StringVar(&c.root, "R", os.Getenv("ZAR_ROOT"), "root directory of zar archive to walk")
+	f.BoolVar(&c.lflag, "l", false, "long form")
 	return c, nil
 }
 
 func (c *Command) Run(args []string) error {
-	if len(args) == 0 {
-		return errors.New("zar ls: no directory specified")
-	}
-	for _, dir := range args {
-		err := archive.Walk(dir, func(zardir string) error {
-			fmt.Println(zardir)
-			return nil
-		})
-		if err != nil {
-			return err
+	if c.root == "" {
+		return errors.New("zar ls: no root specified with -R or ZAR_ROOT")
+	}
+	if len(args) > 1 {
+		return errors.New("zar ls: too many arguments")
+	}
+	var pattern string
+	if len(args) == 1 {
+		pattern = args[0]
+	}
+	return archive.Walk(c.root, func(zardir string) error {
+		printDir(zardir, pattern, c.lflag)
+		return nil
+	})
+}
+
+func fileExists(path string) bool {
+	if path == "-" {
```

Comment:
I don't think you need to check for "-" in this particular case of fileExists. :)